src/main/java/com/ft/methodearticlemapper/messaging/MessageBuilder.java

```diff
@@ -29,18 +30,20 @@
     private final UriBuilder contentUriBuilder;
     private final SystemId systemId;
     private final ObjectMapper objectMapper;
+    private final Map<String, String> contentTypeHeader;
 
-    public MessageBuilder(UriBuilder contentUriBuilder, String systemId, ObjectMapper objectMapper) {
+    public MessageBuilder(UriBuilder contentUriBuilder, String systemId, ObjectMapper objectMapper, Map<String, String> contentTypeHeader) {
         this.contentUriBuilder = contentUriBuilder;
         this.systemId = SystemId.systemIdFromCode(systemId);
         this.objectMapper = objectMapper;
+        this.contentTypeHeader = contentTypeHeader;
     }
 
     Message buildMessage(Content content) {
         MessageBody msgBody = new MessageBody(
                 content,
                 contentUriBuilder.build(content.getUuid()).toString(),
-                RFC3339_FMT.format(OffsetDateTime.ofInstant(content.getLastModified().toInstant(), UTC))
+                RFC3339_FMT.format(OffsetDateTime.ofInstant(content.getLastModified().toInstant(), UTC)), contentTypeHeader
```

Comment:
You need to set one of the values from the map as the value of the `contentType` based on the type of the content. You just inserted the whole map into the MessageBody.